catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/FederationAdminServiceImpl.java

```diff
@@ -66,147 +120,653 @@ public FederationAdminServiceImpl() {
         this.security = security;
     }
 
-    public void init() {
-
+    @Override
+    public String addRegistryEntry(String xml) throws FederationAdminException {
+        Metacard metacard = getRegistryMetacardFromString(xml);
+        return addRegistryEntry(metacard);
     }
 
-    public void setCatalogFramework(CatalogFramework catalogFramework) {
-        this.catalogFramework = catalogFramework;
+    @Override
+    public String addRegistryEntry(String xml, Set<String> destinations)
+            throws FederationAdminException {
+        Metacard metacard = getRegistryMetacardFromString(xml);
+        return addRegistryEntry(metacard, destinations);
     }
 
-    public void setFilterBuilder(FilterBuilder filterBuilder) {
-        this.filterBuilder = filterBuilder;
+    @Override
+    public String addRegistryEntry(Metacard metacard) throws FederationAdminException {
+        return addRegistryEntry(metacard, null);
     }
 
-    public void setParser(Parser parser) {
-        this.configurator =
-                parser.configureParser(Arrays.asList(RegistryObjectType.class.getPackage()
-                                .getName(),
-                        net.opengis.ogc.ObjectFactory.class.getPackage()
-                                .getName(),
-                        net.opengis.gml.v_3_1_1.ObjectFactory.class.getPackage()
-                                .getName()), FederationAdminServiceImpl.class.getClassLoader());
-        this.configurator.addProperty(Marshaller.JAXB_FRAGMENT, true);
-        this.parser = parser;
+    @Override
+    public List<String> addRegistryEntries(List<Metacard> metacards, Set<String> destinations)
+            throws FederationAdminException {
+        validateRegistryMetacard(metacards);
+
+        List<String> registryIds;
+        Map<String, Serializable> properties = new HashMap<>();
+        CreateRequest createRequest = new CreateRequestImpl(metacards, properties, destinations);
+
+        try {
+            CreateResponse response =
+                    security.runWithSubjectOrElevate(() -> catalogFramework.create(createRequest));
+            //loop through to get id's
```

Comment:
Because there are multiple destinations possible, catalog framework will not throw an exception if one of the destination create calls failed. Should check to see if `response.getProcessingErrors()` is not empty and if it isn't then throw an exception. Do the same for the update and delete methods as well.